actions/codecov/action.yml

```diff
@@ -0,0 +1,128 @@
+name: "Codecov"
+description: "Action to upload coverage to Codecov with support for container mode"
+author: hoverkraft
+branding:
+  icon: upload-cloud
+  color: blue
+
+inputs:
+  working-directory:
+    description: |
+      Working directory where coverage files are located.
+      Can be absolute or relative to the repository root.
+    required: false
+    default: "."
+  container:
+    description: "Whether running in container mode (installs dependencies if needed)"
+    required: false
+    default: "false"
+
+runs:
+  using: "composite"
+  steps:
+    # Check and install dependencies for codecov in container mode
+    - name: Check Codecov dependencies
+      if: inputs.container == 'true'
```

Comment:
We dont care if it is container context or not, we check for missing deps in any case